executors/default/pkg/status/polling.go

```diff
@@ -0,0 +1,39 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+// Retry retries the passed retryable function, sleeping for the given backoffDuration
+// If the context times out while executing the retryable function, an error is returned
+func Retry(ctx context.Context, retryable func() error, backoffDuration time.Duration) error {
+	pollingFunc := func(done chan<- bool) {
+		if err := retryable(); err != nil {
+			log.Errorf("Failed to execute the retryable function with: %v", err)
+		}
+		log.Infof("successfully completed the retry function")
+		done <- true
+	}
+	return Poll(ctx, pollingFunc, backoffDuration)
+}
+
+// Poll retries the poller function, sleeping for the given backoffDuration
+// If the poller times out it will return an error
+func Poll(ctx context.Context, poller func(chan<- bool), backoffDuration time.Duration) error {
+	done := make(chan bool)
+	for {
+		go poller(done)
```

Comment:
Doesnt this lead to multiple go routine since it is done in the for loop? Should it be outside the for block?

Reply:
It does, however the max time that the goroutine should run is ~5 seconds based on the runtime of the elements above. Let me add a timer so that we can monitor the timing of runs